backend/webapi/Features/AccessRequests/ImmsBC.cs

```diff
@@ -0,0 +1,108 @@
+namespace Pidp.Features.AccessRequests;
+
+using DomainResults.Common;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+
+using Pidp.Data;
+using Pidp.Infrastructure.HttpClients.Keycloak;
+using Pidp.Infrastructure.HttpClients.Mail;
+using Pidp.Infrastructure.HttpClients.Plr;
+using Pidp.Infrastructure.Services;
+using Pidp.Models;
+using Pidp.Models.Lookups;
+
+public class ImmsBC
+{
+    public static IdentifierType[] AllowedIdentifierTypes => [IdentifierType.PhysiciansAndSurgeons, IdentifierType.Pharmacist, IdentifierType.Nurse];
+    public class Command : ICommand<IDomainResult>
+    {
+        public int PartyId { get; set; }
+    }
+
+    public class CommandHandler : ICommandHandler<Command, IDomainResult>
+    {
+        private readonly IEmailService emailService;
+        private readonly IClock clock;
+        private readonly ILogger<CommandHandler> logger;
+        private readonly IKeycloakAdministrationClient keycloakClient;
+        private readonly PidpDbContext context;
+        private readonly IPlrClient plrClient;
+
+        public CommandHandler(
+            IEmailService emailService,
+            IClock clock,
+            ILogger<CommandHandler> logger,
+            IKeycloakAdministrationClient keycloakClient,
+            PidpDbContext context,
+            IPlrClient plrClient)
+        {
+            this.emailService = emailService;
+            this.clock = clock;
+            this.logger = logger;
+            this.keycloakClient = keycloakClient;
+            this.context = context;
+            this.plrClient = plrClient;
+        }
+
+        public async Task<IDomainResult> HandleAsync(Command command)
+        {
+            var dataObj = await this.context.Parties
+                .Where(party => party.Id == command.PartyId)
+                .Select(party => new
+                {
+                    AlreadyEnroled = party.AccessRequests.Any(request => request.AccessTypeCode == AccessTypeCode.ImmsBC),
+                    UserId = party.PrimaryUserId,
+                    party.Email,
+                    party.DisplayFirstName,
+                    party.Cpn,
+                })
+                .SingleAsync();
+
+            if (dataObj.AlreadyEnroled
+                || dataObj.Email == null
+                || !await this.plrClient.GetStandingAsync(dataObj.Cpn))
```

Comment:
The user not only needs to be in good standing, they need to have a good standing licence from the allowed types. You can do
```C#
  || !(await this.plrClient.GetStandingsDigestAsync(dto.Cpn))
                    .With(AllowedIdentifierTypes)
                    .HasGoodStanding)
```

Reply:
As well, I believe that one of the requirements is that the user has a BC Provider Credential. This requirement should be checked both here and in the `ProfileStatus` for this section